client-app/src/Components/DonatedItemsList.tsx

```diff
@@ -0,0 +1,527 @@
+        }
+    };
+
+    const handleFilterByItemName = (
+        event: React.ChangeEvent<HTMLSelectElement>,
+    ): void => {
+        if (!event.target.value) {
+            setFilteredItems([]);
+            return;
+        }
+        const filtered = donatedItems.filter(
+            item =>
+                item.itemType.toLowerCase() ===
+                event.target.value.toLowerCase(),
+        );
+        setFilteredItems(filtered);
+    };
+
+    const handleFilterByProgram = (
+        event: React.ChangeEvent<HTMLSelectElement>,
+    ): void => {
+        if (!event.target.value) {
+            setFilteredItems([]);
+            return;
+        }
+        const programId = parseInt(event.target.value);
+        const filtered = donatedItems.filter(
+            item => item.programId === programId,
+        );
+        setFilteredItems(filtered);
+    };
+
+    const handleFilterByStatus = (
+        event: React.ChangeEvent<HTMLSelectElement>,
+    ): void => {
+        if (!event.target.value) {
+            setFilteredItems([]);
+            return;
+        }
+        const filtered = donatedItems.filter(
+            item =>
+                item.currentStatus.toLowerCase() ===
+                event.target.value.toLowerCase(),
+        );
+        setFilteredItems(filtered);
+    };
+
+    const toggleAssignProgram = (): void => {
+        setAssignProgramClicked(!assignProgramClicked);
+    };
+
+    const handleAddNewDonationClick = (): void => {
+        navigate('/adddonation');
+    };
+    const downloadBarcode = (id: number) => {
+        const barcodeElement = document.getElementById(`barcode-${id}`);
+        if (barcodeElement) {
+            html2canvas(barcodeElement)
+                .then(canvas => {
+                    const image = canvas.toDataURL('image/png');
+                    const link = document.createElement('a');
+                    link.href = image;
+                    link.download = `barcode-${id}.png`;
+                    document.body.appendChild(link);
+                    link.click();
+                    document.body.removeChild(link);
+                })
+                .catch(err =>
+                    console.error('Error downloading the barcode: ', err),
+                );
+        } else {
+            console.error('Barcode element not found');
+        }
+    };
+
+    if (loading) {
+        return <div>Loading...</div>;
+    }
+
+    if (error) {
+        return <div>Error: {error}</div>;
+    }
+
+    return (
+        <div>
+            <div className="header">
+                <div className="logo-container">
+                    <img
+                        src="https://www.bworks.org/wp-content/themes/bworks/library/images/logo-bworks.png"
+                        alt="BWorks Logo"
+                        className="logo"
+                    />
+                </div>
+                <div className="options">
```

Comment:
Consider extracting this button into a separate reusable component to reduce code duplication and improve maintainability.